shortfin/python/shortfin_apps/llm/components/generate.py

```diff
@@ -195,16 +196,32 @@ async def run(self):
                     if self.gen_req.is_single
                     else self.gen_req.sampling_params[index]
                 )
+                # Get a fiber from the main_fiber_pool.
+                fiber = None
+                if len(self.service.main_fiber_pool) > 0:
+                    fiber = self.service.main_fiber_pool.pop()
```

Comment:
Lets make a `SfFiberPool` async class. It can be pretty simple, just an `asyncio.Queue` where during initialization we push all of the fibers into it. Then we can do an `await queue.get()` to grab an available thread before running.

This can be defined in its own `pool.py` class with a separate `pool_test.py` for managing fibers and prove threads can run individually.

Reply:
@rsuderman pushing an `sf.Fiber` into an `asyncio.Queue` will require serialization of the fiber, for which there is no existing implementation.

I have created a new `FiberPool` class which wraps a bunch of utility methods to manage the fibers. We still need to work out how to handle a shortage of available fibers, as the current implementation simply creates a new fiber/worker pair if there are none available.